### ISSUE TITLE
Handling intermediate terms with MIREOT extract

### ISSUE THREAD

**Issue by rctauber** [ontodev/robot#254]
When you MIREOT in a term (or terms), sometimes you end up with a long lineage of single parents to a single node (intermediate terms). It would be beneficial to "squish" this lineage to have the closest single ancestor and the bottom child.

This feature would be similar to Ontofox's `includeComputedIntermediates` option.

I was thinking it would look something like:
```
robot extract --method MIREOT --input foo.owl --upper-term bar:001 --lower-term bar:002 --squish
```
(alternatively, we could call it `--computed-intermediates`)

In addition to that, it would be beneficial to have an option that allows something like `includeNoIntermediates`... currently, if you only specify a `--lower-term`, ROBOT will extract everything up to the top-level.

We could either change this so that specifying a `--lower-term` without an `--upper-term` will only extract the single term, or add another option for no intermediates.

**Comment by jamesaoverton** [comment]
This would be good to have.

I prefer options to have arguments, so I suggest an `--intermediates` option with `none`, `all`, `computed` (or `minimal` or something) values.

> We could either change this so that specifying a --lower-term without an --upper-term will only extract the single term, or add another option for no intermediates.

These are distinct:

1. `--upper-term foo --lower-term foo` will currently extract `foo` term without parents; we could improve on this
2. `--upper-term foo --lower-term bar --intermediates none` would assert that `bar` is a child of `foo` (as long as `foo` is an ancestor of `bar`)

**Comment by rctauber** [comment]
That sounds good to me

I suppose if you did `--lower-term foo --intermediates none` it would just grab `foo` as well, which would be an improvement to 1.

**Comment by agbeltran** [comment]
Hi, I'm interested in this feature. Do you have a timeline for this? Thanks

**Comment by rctauber** [comment]
The `--intermediates none` and `--intermediates all` options are working now.

`--intermediates minimal` is currently implemented by `RemoveOperation`, and I'd like to move this to `MireotOperation`. There are some unnecessary steps, and I think performance could be improved. Also, anonymous superclasses are not extracted properly so that will be fixed with the change.

**Comment by agbeltran** [comment]
Great, thank you! Am I right that I need to pull from your repo, though, as the commit b60f7a6 is there? I will try it out.

**Comment by rctauber** [comment]
Yup! Just download and build from https://github.com/rctauber/robot/tree/intermediates

**Comment by agbeltran** [comment]
brilliant, thanks

**Comment by rctauber** [comment]
I have re-updated this feature in a new branch: https://github.com/rctauber/robot/tree/trim

Just keeping track of it here.

**Comment by rctauber** [comment]
Implemented by #441
